egs2/zeroth_korean/asr1/README.md

```diff
@@ -0,0 +1,39 @@
+<!-- Generated by scripts/utils/show_asr_result.sh -->
+Zeroth project introduces free Korean speech corpus and aims to make Korean speech recognition more broadly accessible to everyone. 
+
+The database can be also downloaded from openslr:
+http://www.openslr.org/40
+
+The database is licensed under Attribution 4.0 International (CC BY 4.0)
+
+For more details about Zeroth project, please visit:
+https://github.com/goodatlas/zeroth
+
+# RESULTS
+## Environments
+- date: `Thu Aug  6 05:13:08 KST 2020`
+- python version: `3.7.3 (default, Mar 27 2019, 22:11:17)  [GCC 7.3.0]`
+- espnet version: `espnet 0.7.0`
+- pytorch version: `pytorch 1.2.0`
+- Git hash: `f2bd4912e32735487182c6f0cfddbf30e569a476`
+  - Commit date: `Thu Aug 6 08:31:46 2020 +0900`
+
+## asr_train_asr_transformer_fbank_pitch_bpe
+### WER
+
+|dataset|Snt|Wrd|Corr|Sub|Del|Ins|Err|S.Err|
+|---|---|---|---|---|---|---|---|---|
+|inference_test_clean_inference_asr_model_valid.acc.best|457|6641|97.4|1.5|1.2|0.2|2.9|14.4|
+
```

Comment:
Thank you for contributing! I didn't know this corpus and seems useful corpus.

This result looks good, is it expected result?

Reply:
I am happy to contribute to the ESPNet community. Yes, it is expected results. However, it should be noted that the model is trained using 4 GPUs.

Reply:
Hi,

This model outperforms by a huge margin the best model from Kaldi (2.9 vs 9.45), is that expected? I'm not familiar with the recipe nor korean langage, sorry.
Also, looking at kaldi's RESULTS it seems scoring is done on a different number of words (6641 here vs 9253 in Kaldi, https://github.com/kaldi-asr/kaldi/blob/master/egs/zeroth_korean/s5/RESULTS), is there a reason?

Reply:
Yes, exactly.  
In the kaldi recipe,  a word  is first converted to a sequence of morpheme using the Morfessor.
So, the number of words (in actual the number of morphemes in kaldi) are different between the kaldi and espnet. 

Here are the first two lines of "text" of the test set for kaldi and espnet.
The parts marked in bold are different. You can see that words are more segmented in kaldi .
There are 43 words in kaldi but 32 words in espnet for the same lines.

--------------kaldi-----------------------------------
104_003_0019 지난해 삼 월 김 전 장관의 **동료 인** **장동 련** 홍익대 교수가 민간 **자문단 장으로** **위촉 되면서** 본격적인 **공모 와** 개발 작업에 들어갔다 
104_003_0193 그 **바람 에** **나 의** **몸 도** 겹쳐 쓰러지 며 한창 피어난 노란 동백꽃 속으로 폭 파묻혀 버렸다

--------------espnet-----------------------------------
104_003_0019 지난해 삼 월 김 전 장관의 **동료인** **장동련** 홍익대 교수가 민간 **자문단장으로** **위촉되면서** 본격적인 **공모와** 개발 작업에 들어갔다
104_003_0193 그 **바람에** **나의** **몸도** 겹쳐 쓰러지며 한창 피어난 노란 동백꽃 속으로 폭 파묻혀 버렸다

Reply:
Thanks for the explanation! In that case, I guess it would be better to explain the difference in RESULTS then?

Reply:
Okay, I will update README to explain why the RESULTS are different.

Reply:
I'm  also  not familiar with korean lang. Just to make sure, why didn't you use Morfessor? Or, do you know the reason why Kaldi uses it?

Reply:
I don't think that morpheme is the best sub-word unit for Korean end-to-end ASR. 

Until Mike schuter proposed WPM in the paper "Japanese and korean voice search", morpheme was considered to the best sub-word unit to handle OOV as well as to consider contextual information in Korean ASR based on its linguistic characteristics such as agglutinative language. I think that kaldi uses the Morfessor for that reason.

However, as you know, in most end-to-end ASR framework, straightforward sub-word units having less linguistic prior such as grapheme, syllable or BPE are more common, and there is a report that end-to-end Korean ASR using syll-subword (6k) which is almost BPE shows the best performance on this zeroth-korean domain in the paper "https://arxiv.org/pdf/1910.11590.pdf". The paper reports 2.3% CER, and 2.4% WER.

Reply:
Thank you, but I'm not still sure. 

Maybe people want to compare the result as fair as possible, so if  Morfessor is the standard way for this corpus, it's might be better to consider to apply it for scoring. 

This is just a problem how we define the words, isn't it? We can still use BPE after Morfessor segmentation.  Additionally, I understood that the paper did such processes as it said "We used the morphologically segmented text by using given morfessor model. ". Shouldn't we really use it?

Reply:
I agree that some people may want to compare the result using Morfessor. But, the problem is actually how to define WER.

In espnet recipe, WER means WER because it recovers words back from BPE sequences.

However, in kaldi, WER actually means morpheme error rate. 
Kaldi recipe uses morphemes  as if words after segmenting using Morfessor,  and measures error rate for recognized morphemes instead of decoding back to word sequence.

It causes the recognition difference. I think the WER metric of espnet seems more reasonable because it compares recognized results with the original transcription. I think it would be better that kaldi reports WER by comparing recognition results with original transcription. 

Anyway, I agree that it is useful to provide a recipe using Morfessor. So, I will update the recipe.

Thanks.

Reply:
I update the recipe to use Morfessor as default and results. The WER is 2.3% which is even better than the previous result.

Thanks.

Reply:
Thank you! This result seems reasonable.